sos/collector/sosnode.py

```diff
@@ -762,12 +764,35 @@ def execute_sos_command(self):
         try:
             path = False
             checksum = False
+            config_file_arg = ''
+            if self.opts.node_config_file:
+                config_file_arg = f'--config-file={self.opts.node_config_file}'
+            elif self.opts.inherit_config_file:
+                if not self.local:
+                    remote_config = os.path.join(self.tmpdir,
+                                                 'remote_sos.conf')
+                    self.run_command(f"mkdir -pv {self.tmpdir}",
+                                     timeout=10)
+                    self._transport.copy_file_to_remote(
+                        self.opts.inherit_config_file,
+                        remote_config)
+                    config_file_arg = f'--config-file={remote_config}'
+                else:
+                    config_file_arg = (
+                        f'--config-file={self.opts.inherit_config_file}')
+            self.sos_cmd = (
+                f"{self.sos_cmd} {config_file_arg}"
+                if config_file_arg else self.sos_cmd)
             res = self.run_command(self.sos_cmd,
                                    timeout=self.opts.timeout,
                                    use_shell=True,
                                    need_root=True,
                                    use_container=True,
                                    env=self.sos_env_vars)
+            if self.opts.inherit_config_file and not self.local:
+                self.run_command(f"rm {remote_config} ; rmdir {self.tmpdir}",
+                                 use_shell=True,
+                                 need_root=True)
```

Comment:
Yeah, same here. Very much not a fan of doing `rm`s like this - the local sos installation on the remote node should take care of any cleanup. If we place the inherited config files in `/tmp`, then they would be both out of the way and cleaned up separately by long-standing behavior.

Reply:
I'll presume you mean leaving the inherited conf in the tmp directory and letting `systemd-tmpfiles-clean.timer` RHEL, or `tmpfiles.d` for Ubuntu, handle the cleanup on schedule?

It works for my use case, as long as we are comfortable that the file remains might (although probably uncommon) store `plugin_options` in the file for things like `mysql.dbpass`, `postgresql.password`, `skydive.password`?